azurerm/resource_arm_key_vault.go

```diff
@@ -181,6 +182,18 @@ func resourceArmKeyVault() *schema.Resource {
 				Optional: true,
 			},
 
+			"enabled_for_soft_delete": {
+				Type:         schema.TypeBool,
+				Optional:     true,
+				ValidateFunc: validate.BoolIsTrue(),
```

Comment:
given the code below ignores this if it's not set to `true` - I think we can safely remove this validation function? This allows users to define this variable optionally using a variable e.g.

```
resource "azurerm_key_vault" "test" {
  count = 2
  enabled_for_soft_delete = "${count.index == 0 ? true : false}"
}

Reply:
given the comment in the docs `Once enabled you can not disable this setting anymore!` can we add `ForceNew: true` here?

Reply:
I tried to use ForceNew, but changing from soft delete true to soft delete false yields following error: 
```azurerm_key_vault.test: Error updating Key Vault "vault5353625459524355837" (Resource Group "acctestRG-5353625459524355837"): keyvault.VaultsClient#CreateOrUpdate: Failure sending request: StatusCode=409 -- Original Error: autorest/azure: Service returned an error. Status=<nil> Code="ConflictError" Message="Exist soft deleted vault with the same name. "```
On the other hand, switching soft delete on works in place without issue. 

I thought about adding the purge to the delete function, but the PurgeDeleted function also has some limitations: https://docs.microsoft.com/en-us/azure/key-vault/key-vault-ovw-soft-delete. So this would potentially break as well.

I suspect that removing the validation function and adding `ForceNew: true` could not work due to the way soft deleted key vaults are handled on Azure side.

Reply:
> I thought about adding the purge to the delete function, but the PurgeDeleted function also has some limitations: https://docs.microsoft.com/en-us/azure/key-vault/key-vault-ovw-soft-delete. So this would potentially break as well.

in which case I think we'd want to purge the Key Vault if it exists at deletion time, via an opt-in flag? If users don't opt-into that and delete a soft-deleted KeyVault then they'll get this error, at which point it's up to them to recover the vault and import it into Terraform's Statefile to be able to proceed. What do you think?

We have a similar opt-in property in the VM resource where we optionally delete the OS and Data Disks when the VM is deleted.

Reply:
This sounds like a good idea 👍 I added an option to toggle the purge behaviour and also updated the tests support this.